### PR TITLE
Add users, user authentication

### PR DESCRIPTION
**Pull request by mileszs**
I used [this](http://nithinbekal.com/posts/phoenix-authentication/) as a reference. Its first comment was from @stevegrossi, which I thought was pretty grand.

I left off registration, as I don't want users to register for the site. The end goal is to have admins that we create manually, and job posters who can only "login" via the token-ized URLs in their inboxes.

I was able to load up a console (`iex -S phoenix.server`) and run the following to create a user:

``` elixir
changeset = ElixirJobBoard.User.changeset(%ElixirJobBoard.User{}, %{ "email" => "miles@mileszs.com", "password" => "password")

ElixirJobBoard.Registration.create(changeset, ElixirJobBoard.Repo)
```

Then, I could log in using those credentials. Whee!

PS - I know, we still haven't written any tests. In this experimental phase, I'm okay with that. We should totally write tests before this thing goes live, though. :-) We should probably write tests before we start polishing / writing complicated features. I've already looked at [a code coverage tool](http://tech.findmypast.com/elixir-maintainability/#excoveralls) and [a code linter](http://tech.findmypast.com/elixir-maintainability/#excoveralls).

### CODE REVIEW

**Comment by mileszs** [comment]
Ended up adding some seeds for user accounts to play with in dev. One of those was added just for you, @andrewariley87. I hope you appreciate it.

**Comment by andrewariley87** [comment]
#### Should this be a concern with using comeonin

screen shot 2016-09-26 at 11 39 40 pm

File: priv/repo/seeds.exs [hunk 1]
```diff
@@ -9,3 +9,25 @@
...
+  %ElixirJobBoard.User{},
+  %{ "email" => "pg13@pacers.com", "password" => "password" }
+)
+myles_changeset = ElixirJobBoard.User.changeset(
+  %ElixirJobBoard.User{},
+  %{ "email" => "myles@pacers.com", "password" => "password" }
+)
+zeisloft_changeset = ElixirJobBoard.User.changeset(
+  %ElixirJobBoard.User{},
+  %{ "email" => "nick.zeisloft.iu@pacers.com", "password" => "password" }
```
Comment (andrewariley87) [inline priv/repo/seeds.exs]: smh

**Comment by andrewariley87** [comment]
👍  with comments

**Comment by mileszs** [comment]
@andrewariley87 Updated comeonin. Does it resolve your warning?

**Comment by andrewariley87** [comment]
@mileszs Yes the update got rid of the error but one of the tests is failing.
screen shot 2016-09-27 at 10 04 54 pm

**Comment by andrewariley87** [comment]
@mileszs I added a commit to fix the failing test.

I am ready to merge this in if you are.

**Comment by mileszs** [comment]
I added a test for registration. What do you think, @andrewariley87 ?

**Comment by andrewariley87** [comment]
@mileszs test looks good, Do you want me to merge this PR in?

**Comment by mileszs** [comment]
@andrewariley87 Yes, that'd be great. Let me rebase, here.

**Pull request merged**